ec2/InstanceType.dhall

```diff
@@ -315,16 +316,182 @@ let alternatives =
+                              , InstanceType.m4-4xlarge
+                              , InstanceType.m5-large
+                              , InstanceType.m5-xlarge
+                              , InstanceType.m5-2xlarge
+                              , InstanceType.m5-4xlarge
+                              , InstanceType.m5d-large
+                              , InstanceType.m5d-xlarge
+                              , InstanceType.m5d-2xlarge
+                              , InstanceType.m5d-4xlarge
+                              , InstanceType.m5a-large
+                              , InstanceType.m5a-xlarge
+                              , InstanceType.m5a-2xlarge
+                              , InstanceType.m5a-4xlarge
+                              , InstanceType.m5ad-large
+                              , InstanceType.m5ad-xlarge
+                              , InstanceType.m5ad-2xlarge
+                              , InstanceType.m5ad-4xlarge
+                              , InstanceType.m5n-large
+                              , InstanceType.m5n-xlarge
+                              , InstanceType.m5n-2xlarge
+                              , InstanceType.m5n-4xlarge
+                              , InstanceType.m5dn-large
+                              , InstanceType.m5dn-xlarge
+                              , InstanceType.m5dn-2xlarge
+                              , InstanceType.m5dn-4xlarge
+                              , InstanceType.r5-large
+                              , InstanceType.r5-xlarge
+                              , InstanceType.r5-2xlarge
+                              , InstanceType.r5-4xlarge
+                              , InstanceType.r5a-large
+                              , InstanceType.r5a-xlarge
+                              , InstanceType.r5a-2xlarge
+                              , InstanceType.r5a-4xlarge
+                              , InstanceType.r5n-large
+                              , InstanceType.r5n-xlarge
+                              , InstanceType.r5n-2xlarge
+                              , InstanceType.r5n-4xlarge
+                              ]
+                        
+                        let withcrossinstances = [] : List InstanceType
+                        
+                        in    withcrossinstances
+                            # (       if _params.permitCrossInstanceFamily
+                                
+                                then  crossinstances
+                                
+                                else  [] : List InstanceType
+                              )
+                      )
                 , c5-xlarge =
                       [ InstanceType.c5d-xlarge, InstanceType.c5n-xlarge ]
-                    : List InstanceType
+                    # ( let crossinstances =
+                              [ InstanceType.c5-2xlarge
+                              , InstanceType.c5-4xlarge
```

Comment:
I would rather that we separate the concern of permitting cross-instances (`c5-xlarge` requests also `m5-xlarge`) from the concern of permitting upsizing (`c5-xlarge` requests also `c5-2xlarge`), especially since permitting upsizing likely has a stronger cost effect as the number of permitted spot instance pools grows